### PR TITLE
Consistent AWS Pipeline Configurations - #2184

### LINKED ISSUE #2184: Consistent AWS Pipeline Configurations
**Is your feature request related to a problem? Please describe.**
aws: and aws_* options in yaml are not consistent in the pipeline configuration definitions. 

```
sqs:
  queue_url: "https://sqs.us-east-1.amazonaws.com/123456789012/ApplicationLoadBalancer"
  aws:
    region: "us-east-1"
    sts_role_arn: "arn:aws:iam::123456789012:role/Example-Role"
```
vs
```
- opensearch:
    aws_sts_role_arn: "arn:aws:iam::879571289017:role/rajs-osis-policy-role"
    aws_region: "us-west-2"
    aws_sigv4: true
```

**Describe the solution you'd like**
We should align / have consistent format

### PR DESCRIPTION
**Pull request by kkondaka**
Signed-off-by: Krishna Kondaka <krishkdk@amazon.com>

### Description
Added `aws:` option to specify AWS specific configuration under `opensearch` sink configuration. This is an alternate (and more consistent way) of providing aws configuration.
 
### Issues Resolved
Resolves #2184 
 
### Check List
- [ X] New functionality includes testing.
- [X ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/opensearch/README.md [hunk 1]
```diff
@@ -155,6 +157,15 @@ If a single record turns out to be larger than the set bulk size, it will be sen
 
 - `trace_analytics_service_map`: No longer supported starting Data Prepper 2.0. Use `index_type` instead.
 
+### <a name="aws_configuration">AWS Configuration</a>
+
+The AWS configuration is the same for both SQS and S3.
+
+* `region` (Optional) : The AWS region to use for credentials. Defaults to [standard SDK behavior to determine the region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+* `sts_role_arn` (Optional) : The AWS STS role to assume for requests to SQS and S3. Defaults to null, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
+* `aws_sts_header_overrides` (Optional): A map of header overrides to make when assuming the IAM role for the sink plugin.
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/README.md:166]: We should remove the `aws_` on this property as well.

File: data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java [hunk 2]
```diff
@@ -174,8 +175,22 @@ public static ConnectionConfiguration readConnectionConfiguration(final PluginSe
...
     }
 
-    builder.withAwsSigv4(pluginSetting.getBooleanOrDefault(AWS_SIGV4, false));
-    if (builder.awsSigv4) {
+    Map<String, Object> awsOption = pluginSetting.getTypedMap(AWS_OPTION, String.class, Object.class);
+    boolean awsOptionUsed = false;
+    builder.withAwsSigv4(false);
+    if (awsOption != null && !awsOption.isEmpty()) {
+        awsOptionUsed = true;
+        builder.withAwsSigv4(true);
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/ConnectionConfiguration.java:183]: I rather like this actually - by having the `aws` properties we automatically use SigV4.

File: data-prepper-plugins/opensearch/README.md [hunk 3]
```diff
@@ -155,6 +157,15 @@ If a single record turns out to be larger than the set bulk size, it will be sen
 
 - `trace_analytics_service_map`: No longer supported starting Data Prepper 2.0. Use `index_type` instead.
 
+### <a name="aws_configuration">AWS Configuration</a>
+
+The AWS configuration is the same for both SQS and S3.
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/README.md:162]: I do not think we should include the S3 portion. What if one changes over time? We might forget to update this documentation.

File: data-prepper-plugins/opensearch/README.md [hunk 4]
```diff
@@ -155,6 +157,15 @@ If a single record turns out to be larger than the set bulk size, it will be sen
...
+
+* `region` (Optional) : The AWS region to use for credentials. Defaults to [standard SDK behavior to determine the region](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html).
+* `sts_role_arn` (Optional) : The AWS STS role to assume for requests to SQS and S3. Defaults to null, which will use the [standard SDK behavior for credentials](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials.html).
```
Comment (dlvenable) [inline data-prepper-plugins/opensearch/README.md:165]: This references "SQS and S3" - please correct for OpenSearch service.

**Review by dlvenable — CHANGES_REQUESTED** [review]
Look good thanks! I have a few documentation related comments.

File: data-prepper-plugins/opensearch/README.md (hunk 3 above)
Comment (kkondaka) [reply data-prepper-plugins/opensearch/README.md:162]: I will remove that statement.

File: data-prepper-plugins/opensearch/README.md (hunk 4 above)
Comment (kkondaka) [reply data-prepper-plugins/opensearch/README.md:165]: I will remove that statement.

**Review by dlvenable — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]

**Pull request merged**